File: javascript/rna-transcription/rna-transcription.js
export const toRna = (dna) => {
	const COMPLEMENTS = {
		"G": "C",
		"C": "G",
		"T": "A",
		"A": "U",
	};

  	return dna.split("").map( (nucleotide) => COMPLEMENTS[nucleotide] ).join("");
};

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(dna)
    dna.chars.map { |n| self.transform(n) }.join
  end

  def self.transform(n)
    case n
    when 'C'
      return 'G'
    when 'G'
      return 'C'
    when 'T'
      return 'A'
    when 'A'
      return 'U'
    end
  end
end
